ykrt/src/compile/jitc_yk/aot_ir.rs

```diff
@@ -0,0 +1,201 @@
+//! YK's AOT IR deserialiser.
+//!
+//! This is a parser for the on-disk (in the ELF binary) IR format used to express the
+//! ahead-of-time compiled interpreter.
+//!
+//! The `Display` implementations convert the in-memory data structures into a human-readable
+//! textual format.
+
+use deku::prelude::*;
+use std::{
+    error::Error,
+    ffi::CStr,
+    fmt::{self, Display},
+};
+
+/// A magic number that all bytecode payloads begin with.
+const MAGIC: u32 = 0xedd5f00d;
+/// The version of the bytecode format.
+const FORMAT_VERSION: u32 = 0;
+
+fn deserialise_string(v: Vec<u8>) -> Result<String, DekuError> {
+    match unsafe { CStr::from_ptr(v.as_ptr() as *const i8) }.to_str() {
```

Comment:
Dumb question: if the input is a `Vec`, is that `Vec` the length of the C string? If so, we can avoid the `unsafe` `CStr` stuff?

Reply:
That's right. I think we can just use `String::from_utf8`?

(We could actually zero copy the into a `&CStr`, but I thought it was premature optimisation at this point)

Reply:
>  I think we can just use String::from_utf8

You might need to chop off the nul-byte at the end?

Reply:
You're right. The input is determined by `#[deku(until = ...)]` which says:

> Note: The last value which matches the predicate is read

Which means we do take the null terminator too.

https://docs.rs/deku/latest/deku/attributes/#until

Reply:
Out of curiosity, how would you have done it? Is there a better way than `from_utf8`?

Reply:
Maybe https://doc.rust-lang.org/std/ffi/struct.CStr.html#method.from_bytes_until_nul ?

Reply:
Fixed in [d78198d](https://github.com/ykjit/yk/pull/842/commits/d78198d945a3243269e004101972aa301ae38ac9)